sparrow-py/pysrc/sparrow_py/_result.py

```diff
@@ -65,3 +66,32 @@ def iter_rows(self) -> Iterator[dict]:
             for row in next_batch.to_pylist():
                 yield row
             next_batch = self._ffi_execution.next_pyarrow()
+
+    async def iter_pandas_async(self) -> AsyncIterator[pd.DataFrame]:
+        """
+        Asynchronously iterate over the results as Pandas DataFrames.
+
+        Yields
+        ------
+        pd.DataFrame
+            The next Pandas DataFrame.
+        """
+        next_batch = await self._ffi_execution.next_pyarrow_async()
```

Comment:
Curious on thoughts here.

This (and the other methods) currently rely on a simple method (returning an Option) and then use Python generators to turn that into an iterator.

An alternative would be having the rust code return the iterator itself (by implementing `__next__`/`__anext__`). We could do that, but those get more complicated (eg., https://pyo3.rs/v0.15.0/class/protocols.html#awaitable-objects and needing to return https://pyo3.rs/main/doc/pyo3/pyclass/enum.iteranextoutput). It's totally feasible, but I wonder if it's beneficial?

Even if we wanted the Python result to be iterable directly (eg, `async for expr.run()` would automatically iterate over batches, we can do that by *just* implementing `__aiter__` in Python).

And at the batch level, we shouldn't be crossing Python/Rust so much that the extra layer is too expensive, especially since it keeps the code much more straightforward.

Any thoughts here?

Reply:
This looks good to me. It's simple to comprehend, I can print out batches as they are produced by the generator here, we wouldn't have to introduce more types I don't know about. 

I haven't done much reading on Py03 or the async options we have here. Are there any benefits to returning the iterator that you've since found?

Reply:
I think it would just be letting Rust implement hte Async generator directly, rather than having an Async generator in Python that calls a method in Rust that returns `Option<RecordBatch>` (and is basically the `anext` method). But, it would require more classes (a separate class for each kind of iteration). I may poke at it later, but it unless Python is horribly slow about running the generator, it doesn't seem like it buys us much, and adds complexity to the layer between Python/Rust.